fenix/app/src/main/java/org/mozilla/fenix/components/appstate/AppStoreReducer.kt

```diff
@@ -100,6 +106,9 @@ internal object AppStoreReducer {
                 else -> state.recentSyncedTabState
             },
         )
+        is AppAction.SelectedTabChanged -> state.copy(
```

Comment:
I'm not sure the best way to reconcile this with `OpenTabInBrowser`. We need to be able to react to TabSessionState changes through `SelectedTabChanged` because there are things outside the Redux model that can cause updates, like `TabsUseCases`. 

However, eventually `OpenTabInBrowser` will also be kicking off those kind of state changes, which means we'd hit the reducer twice.